src/main/scala/play/modules/reactivemongo/json.scala

```diff
@@ -37,8 +37,8 @@ object BSONFormats {
 
   implicit object BSONDoubleFormat extends PartialFormat[BSONDouble] {
     val partialReads: PartialFunction[JsValue, JsResult[BSONDouble]] = {
-      case JsNumber(f)                               => JsSuccess(BSONDouble(f.toDouble))
-      case JsObject(("$double", JsNumber(v)) +: Nil) => JsSuccess(BSONDouble(v.toDouble))
+      case JsNumber(f)                             => JsSuccess(BSONDouble(f.toDouble))
+      case JsObject(Seq(("$double", JsNumber(v)))) => JsSuccess(BSONDouble(v.toDouble))
```

Comment:
In our envirionment (play 2.4), JsObject(Seq(XX, XXX)) failed to patten match, so everything with JsObject goes to the default BSONDocument format